tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy

```diff
@@ -0,0 +1,93 @@
+import org.sonatype.nexus.common.app.GlobalComponentLookupHelper;
+import org.sonatype.nexus.repository.maintenance.MaintenanceService;
+import org.sonatype.nexus.repository.storage.ComponentMaintenance;
+import org.sonatype.nexus.repository.storage.Query;
+import org.sonatype.nexus.script.plugin.RepositoryApi;
+import org.sonatype.nexus.script.plugin.internal.provisioning.RepositoryApiImpl;
+import com.google.common.collect.ImmutableList;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+
+def retentionDays = 15;
+def retentionCount = 15;
+def tagList = [];
+def repositoryName = 'BartsDockerRepo';
+def whitelisted_tag_suffixes = ["queens", "ocata", "newton", "latest", "5.0-40", "5.0-94", "5.0-122", "5.0-129", "5.0-161", "5.0-168", "5.0-214", "5.0-309", "5.0-360"].toArray();
+log.info(":::Cleanup script started!");
+MaintenanceService service = container.lookup("org.sonatype.nexus.repository.maintenance.MaintenanceService");
+def repo = repository.repositoryManager.get(repositoryName);
+def tx = repo.facet(StorageFacet.class).txSupplier().get();
+def components = null;
+try {
+    tx.begin();
+    components = tx.browseComponents(tx.findBucket(repo));
+} catch (Exception e) {
+    log.info("Error: " + e);
+} finally {
+    if (tx != null)
+        tx.close();
+}
+
+if (components != null) {
+    def retentionDate = DateTime.now().minusDays(retentionDays).dayOfMonth().roundFloorCopy();
+    int deletedComponentCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+
+// comp.version() expected to be in a format {{ os }}-{{ realse }}-{{ branch }}-{{ build }} e.g. rhel-queens-master-386 
+
+    listOfComponents.reverseEach { comp ->
+        def tag = comp.version();
+        def tagSplited = tag.split("-");
+        log.info("tagsplit ${tag} ${tagSplited}");
+        def build_number = tagSplited[tagSplited.length - 1];
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + tagSplited[tagSplited.length - 1];
+        log.info("new two parts ${two_parts_build_number}");
+        if (whitelisted_tag_suffixes.contains(build_number)) {
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
+        } else if (whitelisted_tag_suffixes.contains(two_parts_build_number)){
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
+        }else {
+            if (tagList.count(build_number.toInteger()) == 0) {
+                tagList.add(build_number.toInteger());
+                println tagList.sort();
+            }
+        }
+    }
+
+    listOfComponents.reverseEach { comp ->
+        def tag = comp.version();
+        def tagSplited = tag.split("-");
+        def build_number = tagSplited[tagSplited.length - 1];
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + tagSplited[tagSplited.length - 1];
+        def retentionList = tagList;
+        if(tagList.size() > retentionCount){
```

Comment:
Think you can move this outside, before the 'each' closure as this if/else should be run only once, to either trim the list or just return